packages/core/logging/core-logging-common-internal/src/layouts/conversions/message.test.ts

```diff
@@ -62,4 +62,37 @@ describe('MessageConversion', () => {
       '\\u001b\\u0000[31mESC-INJECTION-LFUNICODE:\\u001b[32mSUCCESSFUL\\u001b[0m\\u0007\n\nInjecting 10.000 lols 😂\\u001b[10000;b\\u0007'
     );
   });
+
+  test('it should encode/escape ANSI chars lines from the message when not a string', () => {
+    expect(
+      MessageConversion.convert(
+        {
+          ...baseRecord,
+          // @ts-expect-error message is supposed to be a string
+          message: {
+            toString: () => 'toString...\u001b[5;7;6mThis is Fine\u001b[27m',
+          },
+        },
+        false
+      )
+    ).toEqual('toString...\\u001b[5;7;6mThis is Fine\\u001b[27m');
+  });
+
+  test('it should encode/escape ANSI chars lines from the error stack when not a string', () => {
+    expect(
+      MessageConversion.convert(
+        {
+          ...baseRecord,
+          message: 'Some message that will be ignored',
+          error: {
+            // @ts-expect-error message is supposed to be a string
+            stack: {
+              toString: () => 'stackToString...\u001b[5;7;6mThis is Fine\u001b[27m',
+            },
+          },
+        },
+        false
+      )
+    ).toEqual('stackToString...\\u001b[5;7;6mThis is Fine\\u001b[27m');
+  });
```

Comment:
nit: can we add one test where we provide an actual Error object with a custom stack trace?

Reply:
As discussed on slack, the test is fairly close to the real world scenario:

https://github.com/elastic/kibana/blob/0ad9abc5acb496a2f8d4c397a527c245cd4b6ca7/x-pack/plugins/observability/common/utils/unwrap_es_response.ts#L19-L34